aad/aad-b2c-with-spring-security/README.adoc

```diff
@@ -0,0 +1,279 @@
+
+==== Register your Azure Active Directory B2C application
+
+We can create another application to show the feature of validate audience.
+
+Follow the guide of https://docs.microsoft.com/azure/active-directory-b2c/tutorial-register-applications[AAD B2C application registry].
+Please ensure that your b2c application's `Redirect URL` is configured to `http://localhost:8080/login/oauth2/code/`.
+In order to jump to the correct page (home) after `profile-edit` and `password-reset`, we need to add `http://localhost:8080/` in the redirect uri too.
+You can add additional `Redirect URL` with https://docs.microsoft.com/azure/active-directory/develop/quickstart-register-app#add-a-redirect-uri[Add Redirect URL].
+
+==== Fill necessary values in sample project.
+
+1. Fill necessary values in `application.yml`, like `${your-jwk-set-uri}`.
+
+==== Boot up the application
+
+Lunch the sample project. Then open `http://localhost:8080/resourceServer` by browser.
+Choose `web-application` and sign in. Which will access resource server success.
+Access `http://localhost:8080/logout` to sign out and access `http://localhost:8080/resourceServer` again.
+Choose `web-application2` and sign in. Which will access resource server fail.
+
+[[Sample-09-resource-server-authorize-by-scope]]
+=== Sample 09: Resource server authorize by scope
+
+Sample project: <<./sample-09-resource-server-authorize-by-scope/README.adoc#chapter-link, sample-09-resource-server-authorize-by-scope>>
+
+Compared with <<Sample-07-Resource-server-validate-audience, Sample-07-Resource-server-validate-audience>>, this sample updates the `SampleController.java` and `WebSecurityConfiguration.java`.
+
+==== Fill necessary values in sample project.
+
+1. Fill necessary values in `application.yml`, like `${your-jwk-set-uri}`.
+
+==== Boot up the application
+
+We need to get related access token before we access this resource server, we can get related access token from previous sample, please refer to <<Sample-10-web-application-access-resource-server-validate-scope, Sample-10-web-application-access-resource-server-validate-scope>>.
+Lunch the sample project. This resource server will validate the accessToken's `roles` claim.
+
+[[Sample-10-web-application-access-resource-server-validate-scope]]
+=== Sample 10: Web application access resource server validate scope
+
+Sample project: <<./sample-10-web-application-access-resource-server-validate-scope/README.adoc#chapter-link, sample-10-web-application-access-resource-server-validate-scope>>
+
+Compared with <<Sample-08-web-application-access-resource-server-validate-audience, Sample-08-web-application-access-resource-server-validate-audience>>, this sample updates the `application.yml`.
+
+==== Fill necessary values in sample project.
+
+1. Fill necessary values in `application.yml`, like `${your-jwk-set-uri}`.
+
+==== Boot up the application
+
+Lunch the sample project. Then open `http://localhost:8080/resourceServerForScope1` by browser.
```

Comment:
nit: How about using URL like this: `http://localhost:8080/client-1/resource-server-1/scope-1`, same to other URLs.